File: B-Job_Assignment.rb
def Jobs_Finish_Time(n, jobs_time)
    ans = 10000000
    for i in 0..n - 1
        for j in 0..n - 1
            if i == j
                ans = [ans, jobs_time[i][0] + jobs_time[j][1]].min
            else
                ans = [ans, [jobs_time[i][0], jobs_time[j][1]].max].min
            end
        end
    end
    return ans
end

n = gets.to_i
jobs_time = []
for i in 0..n - 1
    person_time = gets.split(' ').map(&:to_i)
    jobs_time.push(person_time)
end

ans = Jobs_Finish_Time(n, jobs_time)
puts ans

File: B-Job_Assignment.cpp
#include <iostream>
#include <vector>
using namespace std;


int main(){
    int n;
    cin >> n;
    vector<vector<int> > jobs_time(n, vector<int>(2));


    for (int i = 0; i < n; i++){
        for (int j = 0; j < 2; j ++){
            cin >> jobs_time.at(i).at(j);
        }
    }
    int jobs_finish_time = 10000000;

    for (int i = 0; i < n; i ++){
        for (int j = 0; j < n; j ++){
            if (i == j){
                jobs_finish_time = min(jobs_finish_time, (jobs_time.at(i).at(0) + jobs_time.at(j).at(1)));
            }else {
                jobs_finish_time = min(jobs_finish_time, (max(jobs_time.at(i).at(0), jobs_time.at(j).at(1))));
            }
        }
    }
    cout << jobs_finish_time << endl;
}

